### DISCUSSION TITLE
MQTT: Is car charging

### DISCUSSION THREAD

**Discussion by gurmukhp** [teslamate-org/teslamate#3725] [General]
Does anyone know how to determine if the car is currently charging or not? I have looked through the MQTT docs, but there doesn't seem to be anything obvious, do we need to hash a few MQTT topics together or something?

thanks!

**Comment by eliluong** [comment]
![image](https://github.com/teslamate-org/teslamate/assets/15806938/dc8801a0-a70c-4f0c-8f75-af8efa922d22)

This says it has a `charging` state topic.

**Comment by brianmay** [comment]
There is also the "teslamate/cars/{id}/charging_state" topic, which gives Starting, Complete, Charging, Disconnected, and NoPower.

Hope I got that right :-)

This one may not have reached the docs, probably should get added.

**Comment by gurmukhp** [comment]
This is perfect, thank you both! I was searching the topics beginning with 'charge'.
